.github/workflows/deploy_dev_containers.yml

```diff
@@ -0,0 +1,51 @@
+name: (dev) Docker image build and publish
+on:
+  workflow_dispatch:
+    inputs:
+      instrument_name:
+        description: 'Instrument name'
+        required: true
+        type: string
+      git_url:
+        description: 'IMAP Processing Git URL to build from'
+        required: true
+        type: string
+        default: 'https://github.com/IMAP-Science-Operations-Center/imap_processing.git@dev'
+
+# concurrency required to avoid lock contention during ECR provisioning
+concurrency: ci-${{ github.repository }}-${{ inputs.instrument_name }}-dev-docker-pipeline
+
+jobs:
+  deploy_containers:
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
+
+      # https://github.com/aws-actions/configure-aws-credentials
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::449431850278:role/GitHubDeploy
+          aws-region: us-west-2
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v2
+        with:
+         mask-password: "true" # see: https://github.com/aws-actions/amazon-ecr-login#docker-credentials
+
+      - name: Build, tag, and push docker image to Amazon ECR
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          REPOSITORY: ${{ inputs.instrument_name}}
+          IMAGE_TAG: latest
```

Comment:
I definitely agree with this! I didn't want to mess with it for now, but we should move to this version of things. In fact, maybe the workflow should be different and we add another workflow job that would just move tags from "dev"/"latest" -> "prod" when we verify it is working as expected rather than rebuilding the prod container.

Reply:
I captured a note about this which I will create GitHub issues afterwards.